tor/src/main/scala/org/bitcoins/tor/client/TorClient.scala

```diff
@@ -0,0 +1,231 @@
+package org.bitcoins.tor.client
+
+import grizzled.slf4j.Logging
+import org.bitcoins.core.util.EnvUtil
+import org.bitcoins.tor.TorProtocolHandler._
+import org.bitcoins.tor.config.TorAppConfig
+import org.bitcoins.tor.{Socks5ProxyParams, TorParams}
+
+import java.io.{File, FileNotFoundException}
+import java.net.InetSocketAddress
+import java.nio.file.{Files, Path, StandardCopyOption}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.sys.process.{Process, ProcessBuilder}
+import scala.util.{Failure, Success, Try}
+
+/** A trait that helps start bitcoind/eclair when it is started via bitcoin-s */
+class TorClient(implicit ec: ExecutionContext, conf: TorAppConfig)
```

Comment:
nit: `class TorClient()(implicit ec: ExecutionContext, conf: TorAppConfig)`

Reply:
Could we use `NativeProcessFactory` to have some continuity? 

I believe `NativeProcessFactory` can be moved into `core` with no issues 

https://github.com/bitcoin-s/bitcoin-s/blob/4055de7690f6a96d63be225c019243749102f687/bitcoind-rpc/src/main/scala/org/bitcoins/rpc/util/NativeProcessFactory.scala#L9

Reply:
Oh yeah, copy pasted it initially, meant to move it